### PR TITLE
improve_bfc_allocator

### PR DESCRIPTION
**Pull request by Xuxue1**
1.Modify the Roundedbytes method 
2.Modify the frequently called small methods to make them inline methos

### CODE REVIEW

**Review by xannyphvntom — DISMISSED** [review]

**Comment by jsimsa** [comment]
@gbaned I am not familiar with this code. Please add the author of these methods as a reviewer.

**Comment by Xuxue1** [comment]
@mrry 
Can you help to review this code.thanks!

File: tensorflow/core/common_runtime/bfc_allocator.h [hunk 1]
```diff
@@ -372,7 +372,7 @@ class BFCAllocator : public VisitableAllocator {
   size_t BinNumToSize(BinNum index) {
     return static_cast<size_t>(256) << index;
   }
-  BinNum BinNumForSize(size_t bytes) {
+  inline BinNum BinNumForSize(size_t bytes) {
```
Comment (mrry) [inline tensorflow/core/common_runtime/bfc_allocator.h]: This change has no effect because we're in a class definition. Please revert this line.

Comment (Xuxue1) [reply tensorflow/core/common_runtime/bfc_allocator.h]: Thank you very much for your suggestion.

**Comment by Xuxue1** [comment]
@mrry

**Review by mrry — DISMISSED** [review]

File: tensorflow/core/common_runtime/bfc_allocator.cc [hunk 2]
```diff
@@ -229,10 +229,8 @@ void* BFCAllocator::AllocateRaw(size_t unused_alignment, size_t num_bytes,
 }
 
 // static
-size_t BFCAllocator::RoundedBytes(size_t bytes) {
-  size_t rounded_bytes =
-      (kMinAllocationSize *
-       ((bytes + kMinAllocationSize - 1) / kMinAllocationSize));
+inline size_t BFCAllocator::RoundedBytes(size_t bytes) {
```
Comment (mrry) [inline tensorflow/core/common_runtime/bfc_allocator.cc]: The `inline` here is causing link-time failures when building "//tensorflow/core:common_runtime_gpu_gpu_bfc_allocator_test".

**Comment by gbaned** [comment]
@Xuxue1 Could you please check reviewer comments and keep us posted. Thanks!

File: tensorflow/core/common_runtime/bfc_allocator.cc (hunk 2 above)
Comment (Xuxue1) [reply tensorflow/core/common_runtime/bfc_allocator.cc]: I remove it

**Review by mrry — APPROVED** [review]

**Comment by mrry** [comment]
Hi there. Sorry for the long delay on reviewing this PR. After some internal discussion with @dubey, who works more closely on this part of the code, we've decided that the additional code complexity from the bitwise operations outweighs the performance improvement (which is uncertain, since I'd expect an optimizing compiler to find the same optimization when dividing/multiplying by a constant power of two). Therefore we're going to close this PR without accepting it.

If you can measure a performance difference with and without this change, we'd be open to reconsidering.

**Pull request closed without merging**